### PR TITLE
Bump Roslyn to 4.11.0 (CLI, Testing Framework)

### PR DESCRIPTION
**Pull request by josefpihrt**

### CODE REVIEW

**Comment by basverweij-pp** [comment]
Hi @josefpihrt, do you have a timeline for when a new version of the CLI will be published? (Maybe as a pre-release?) We really love your tool, but are unable to use it right now as our solutions have been upgraded to Roslyn 4.10/4.11. (It fails to include source generated code which we use heavily.) Many thanks!

**Comment by josefpihrt** [comment]
Hi @basverweij-pp,

I had a plan to update Roslyn version much earlier. But I discovered that Microsoft unintentionally dropped support for .NET 6 for Roslyn 4.10 and higher (https://github.com/dotnet/roslyn/issues/74016).

As there seems to be no progress on this issue on Microsoft's side I will update Roslyn version to latest version and drop support for .NET 6 (meaning dropping support for installed .NET SDK).

Then, a new version of CLI can be released within several days.

File: src/Directory.Build.props [hunk 1]
```diff
@@ -31,8 +31,8 @@
   
   <PropertyGroup>
     <Version Condition="'$(Version)' == ''">1.0.0</Version>
-    <RoslynatorCliRoslynVersion>4.9.2</RoslynatorCliRoslynVersion>
-    <RoslynatorTestingRoslynVersion>4.9.2</RoslynatorTestingRoslynVersion>
+    <RoslynatorCliRoslynVersion>4.10.0</RoslynatorCliRoslynVersion>
+    <RoslynatorTestingRoslynVersion>4.10.0</RoslynatorTestingRoslynVersion>
```
Comment (josefpihrt) [inline src/Directory.Build.props:35]:
```suggestion
    <RoslynatorCliRoslynVersion>4.12.0</RoslynatorCliRoslynVersion>
    <RoslynatorTestingRoslynVersion>4.12.0</RoslynatorTestingRoslynVersion>
```

File: ChangeLog.md [hunk 2]
```diff
@@ -15,6 +15,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fix analyzer [RCS1018](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1018) ([PR](https://github.com/dotnet/roslynator/pull/1510))
 - Fix analyzer [RCS1264](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1264) ([PR](https://github.com/dotnet/roslynator/pull/1511))
 
+### Changed
+
+- Bump Roslyn to 4.12.0 ([PR](https://github.com/dotnet/roslynator/pull/1483))
```
Comment (josefpihrt) [inline ChangeLog.md:20]:
```suggestion
- Bump Roslyn to 4.11.0 ([PR](https://github.com/dotnet/roslynator/pull/1483))
```

**Comment by josefpihrt** [comment]
@basverweij-pp https://www.nuget.org/packages/Roslynator.DotNet.Cli/0.9.0

**Comment by basverweij-pp** [comment]
> @basverweij-pp https://www.nuget.org/packages/Roslynator.DotNet.Cli/0.9.0

thanks!

**Pull request merged**